src/shadowbox/server/manager_service.ts

```diff
@@ -93,18 +98,18 @@ export class ShadowsocksManagerService {
   public renameServer(req: RequestType, res: ResponseType, next: restify.Next): void {
     const name = req.params.name;
     if (typeof name !== 'string' || name.length > 100) {
-      res.send(400);
-      next();
+      next(new restify.BadRequestError(
+          `Requested server name should be a string <= 100 characters long.  Got ${name}`));
```

Comment:
I wonder if this should be an InvalidArgumentError (409) since the argument itself is incorrect.

Reply:
Done